src/config-file-schema.ts

```diff
@@ -126,8 +124,17 @@ export function ConfigFileSchema<T>(opts: {
     try {
       const parsed = parse(serialized);
       return parsed;
-    } catch (error) {
-      return undefined;
+    } catch (ex: any) {
+      if (ex instanceof SyntaxError) {
+        if (opts.JSONDecodeError) {
+          const message =
+            opts.JSONDecodeError.message ||
+            `Contents of ${path} could not be parsed. Please ensure file is in valid JSON format.`;
```

Comment:
Can we add the text from the original error log here two? It might contain some useful information like a missing comma or something

Reply:
Yes, good point!

Reply:
I've updated the error type and returned the original error message concatenated to the returned message!